group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1084,10 +1129,14 @@ public CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGro
     ) throws ApiException {
         throwIfConsumerGroupHeartbeatRequestIsInvalid(request);
 
-        if (request.memberEpoch() == LEAVE_GROUP_MEMBER_EPOCH) {
+        if (request.memberEpoch() == LEAVE_GROUP_MEMBER_EPOCH || request.memberEpoch() == -2) {
```

Comment:
nit: Should we introduce a constant for -2 as well?

Reply:
done